flax/linen/module.py

```diff
@@ -344,10 +356,20 @@ def reimport(self, other):
     '__reduce__', '__reduce_ex__', '__copy__', '__deepcopy__')
 
 
-_caches = weakref.WeakKeyDictionary()
+tuple_reduce = lambda xs, x: xs + (x,)
+tuple_init = lambda: ()
+
+
+capture_call_intermediates = lambda _, method_name: method_name == '__call__'
+
 
 # Base Module definition.
 # -----------------------------------------------------------------------------
+
+
+_caches = weakref.WeakKeyDictionary()
```

Comment:
Can you add a comment explaining why we need weak references here?

Reply:
Oops I accidentally switched this code around. Doesn't actually belong to the PR